GCTC-NTGC/gc-digital-talent
# Bug - 4310 - Add cache for talent search forms

Resolves #4310 

## 📋  Summary

This refactors some forms and adds `cacheKey` to all talent search forms so no data is lost on page refresh.

## ❗ Important

We are now styling inputs based on the current form state.

### What does this mean?

We must now wrap **_most_** (assume all) form components in the `react-hook-form` `FormProvider`. This is so the form components can access the state and render the field accordingly. This will help with styling the following states:

- Unsaved (this PR)
- Invalid
- Valid
- Base

## ✅  To Do

- [x] Track dirty form values
- [x] Unsaved changes summary

## 🧪 Testing

1. Build talent search `npm run production --workspace=talentsearch`
2. Navigate to all profile forms and test the folllowing:
    1. Edit the form
    2. Refresh page
    3. Ensure the edited state appears
3. Run tests `npm run test --workspace=talentsearch`
